issue-259.md

```diff
@@ -1,52 +1,92 @@
 # Sub-Issue 07: Integration tests and CI wiring
 
 ## Summary
-Add end-to-end tests that validate resctrl plugin behavior for Add/Remove events, state transitions (Success/Partial/Failure), startup cleanup, and retries. Keep container runtime interaction and PID enumeration real, and mock only the resctrl filesystem via the `FsProvider` in the `resctrl` crate so tests run even on hosts without resctrl support.
+Add end-to-end tests that validate resctrl plugin behavior for Add/Remove events, pod group state and per-pod reconciliation counts, startup cleanup, and retries. Align tests with the implemented event model and APIs introduced while working on Issue #252 and its sub-issues:
+- Events: `PodResctrlEvent::{AddOrUpdate(PodResctrlAddOrUpdate), Removed(PodResctrlRemoved)}`
+- Pod group state: `ResctrlGroupState::{Exists(String), Failed}`
+- Per-pod counters: `{ total_containers, reconciled_containers }`
+
+Default integration tests run with a mocked resctrl filesystem provider and a test PID source for determinism. Hardware E2E runs on EC2 use the real resctrl filesystem and real PID enumeration.
 
 ## Scope
-- Add an integration test crate (similar to `crates/nri/tests`) covering:
-  - Startup with preexisting pods/containers: events emitted after reconcile.
-  - Container add/update: event dedup and state changes.
-  - ENOSPC on first attempt → Failure event → retry → Success path.
-  - `cleanup_on_start` removes prefixed groups only.
+- Add an integration test crate (or module) covering:
+  - Startup with preexisting pods/containers: plugin emits `AddOrUpdate` with `group_state = Exists(_) | Failed` and correct `{total,reconciled}` counts after reconcile.
+  - Container add/update: event dedup; counts only change when a container transitions to `Reconciled`.
+  - ENOSPC on first attempt → `AddOrUpdate` with `group_state = Failed` → `retry_group_creation` transitions to `Exists(path)`; follow-up container reconciliation improves counts.
+  - `cleanup_on_start` removes only prefixed groups at resctrl root and root-level `mon_groups` (no traversal into control-group-local `mon_groups`). No pod events emitted for cleanup.
 - Hardware E2E tests on EC2 (real resctrl):
```

Comment:
this type of test seems to still be missing:
- tests for auto_mount=true and auto_mount=false behavior when resctrl is mounted and when resctrl isn't mounted
- cleanup tests (we have a good description and implementation already in #258 )